### PR TITLE
#5090: Update automation-anywhere/oauth2 integration definition

### PR DESCRIPTION
**Pull request by johnnymetz**
## What does this PR do?

- Closes #5090 

## Future Work

- Copy / paste the definition into prod

## Checklist

- [x] Designate a primary reviewer: @twschiller 
- [x] Add tests

### CODE REVIEW

File: contrib/services/automation-anywhere-oauth2.yaml [hunk 1]
```diff
@@ -18,10 +27,10 @@ inputSchema:
 authentication:
   baseURL: "{{{ controlRoomUrl }}}"
   oauth2:
-    authorizeUrl: "https://oauthconfigapp.automationanywhere.digital/client/oauth/authorize?hosturl={{{ controlRoomUrl }}}&audience=https://controlroom"
-    tokenUrl: "https://oauthconfigapp.automationanywhere.digital/client/oauth/token"
+    authorizeUrl: "{{{ authConfigOrigin }}}/client/oauth/authorize?hosturl={{{ controlRoomUrl }}}&audience=https://controlroom"
```
Comment (johnnymetz) [inline contrib/services/automation-anywhere-oauth2.yaml:30]: @twschiller I added `default` values above so isn't the schema smart enough to substitute the default in here if the value is not defined? Or do I need to use [inverted sections](https://mustache.github.io/mustache.5.html#Inverted-Sections) as you mentioned before?

Comment (twschiller) [reply contrib/services/automation-anywhere-oauth2.yaml:30]: @johnnymetz you have to use inverted sections

In JSON Schemas, the `default` is a hint for the initial value in forms. The default isn't actually passed in as the value if one is not provided

File: contrib/services/automation-anywhere-oauth2.yaml [hunk 2]
```diff
@@ -9,6 +9,15 @@ inputSchema:
   $schema: "https://json-schema.org/draft/2019-09/schema#"
   type: object
   properties:
+    authConfigOrigin:
+      type: string
+      description: "The AuthConfigApp origin"
```
Comment (twschiller) [inline contrib/services/automation-anywhere-oauth2.yaml:14]: Matching naming convention: https://docs.automationanywhere.com/bundle/enterprise-v2019/page/enterprise-cloud/topics/control-room/administration/settings/authconfig-app-enable-oauth2.html

**Review by twschiller — CHANGES_REQUESTED** [review]

**Comment by johnnymetz** [comment]
Ready for another review

File: contrib/services/automation-anywhere-oauth2.yaml [hunk 3]
```diff
@@ -2,13 +2,22 @@ apiVersion: v1
 kind: service
 metadata:
   id: "automation-anywhere/oauth2"
-  version: 3.0.0
+  version: 4.0.0
```
Comment (johnnymetz) [inline contrib/services/automation-anywhere-oauth2.yaml:5]: I figured these changes warrant a major version bump. Let me know if you disagree

Comment (twschiller) [reply contrib/services/automation-anywhere-oauth2.yaml:5]: Doesn't matter too much. Technically the changes are backward compatible, so would also be OK with a minor version bump

File: contrib/services/automation-anywhere-oauth2.yaml [hunk 4]
```diff
@@ -18,10 +27,10 @@ inputSchema:
 authentication:
   baseURL: "{{{ controlRoomUrl }}}"
   oauth2:
-    authorizeUrl: "https://oauthconfigapp.automationanywhere.digital/client/oauth/authorize?hosturl={{{ controlRoomUrl }}}&audience=https://controlroom"
-    tokenUrl: "https://oauthconfigapp.automationanywhere.digital/client/oauth/token"
+    authorizeUrl: "{{#authConfigOrigin}}{{{.}}}{{/authConfigOrigin}}{{^authConfigOrigin}}https://oauthconfigapp.automationanywhere.digital{{/authConfigOrigin}}/client/oauth/authorize?hosturl={{{ controlRoomUrl }}}&audience=https://controlroom"
```
Comment (twschiller) [inline contrib/services/automation-anywhere-oauth2.yaml:30]: TIL about the `.`. Where did you find it? I still don't see it in the documentation

File: src/services/factory.test.ts [hunk 5]
```diff
@@ -58,6 +76,26 @@ describe("LocalDefinedService", () => {
...
+
+  test("default client ID", () => {
+    const service = fromJS(
+      automationAnywhereOAuth2 as unknown as ServiceDefinition
+    );
+    const oauth2 = service.getOAuth2Context({} as unknown as ServiceConfig);
+    expect(oauth2.client_id).toBe("g2qrB2fvyLYbotkb3zi9wwO5qjmje3eM");
+  });
+
+  test("custom client ID", () => {
```
Comment (twschiller) [inline src/services/factory.test.ts:88]: ❤️

File: contrib/services/automation-anywhere-oauth2.yaml (hunk 4 above)
Comment (johnnymetz) [reply contrib/services/automation-anywhere-oauth2.yaml:30]: https://stackoverflow.com/a/62030327/6611672

File: contrib/services/automation-anywhere-oauth2.yaml (hunk 3 above)
Comment (johnnymetz) [reply contrib/services/automation-anywhere-oauth2.yaml]: Changed to a minor version bump

File: contrib/services/automation-anywhere-oauth2.yaml (hunk 4 above)
Comment (johnnymetz) [reply contrib/services/automation-anywhere-oauth2.yaml:30]: More context here: https://stackoverflow.com/questions/25320065/what-is-in-mustache

**Pull request merged**